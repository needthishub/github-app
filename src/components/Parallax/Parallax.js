import React, {useState, useEffect} from 'react';
import {
  ParallaxBackgroundImg4, ParallaxBackgroundImg3, ParallaxBackgroundImg2, ParallaxBackgroundImg1, ParallaxSection,
} from './styledComponents';

const Parallax = () => {
  const [offsetY, setOffsetY] = useState(0);
  const handleScroll = () => setOffsetY(window.pageYOffset);

  useEffect(() => {
    window.addEventListener('scroll', handleScroll);

    return () => window.removeEventListener('scroll', handleScroll);
  }, []);

  return (
    <ParallaxSection>
      <ParallaxBackgroundImg1
        style={{transform: `translateY(-${offsetY * 0.3}px)`}}
      />
      <ParallaxBackgroundImg2
        style={{transform: `translateY(-${offsetY * 0.3}px)`}}
      />
      <ParallaxBackgroundImg3
        style={{transform: `translateY(-${offsetY * 0.3}px)`}}
      />
      <ParallaxBackgroundImg4
        style={{transform: `translateY(-${offsetY * 0.3}px)`}}
      />
    </ParallaxSection>
  );
};

export default Parallax;
