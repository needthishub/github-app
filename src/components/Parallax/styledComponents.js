import styled from 'styled-components';
import img38582 from '../../images/38582.jpg';
import img38593 from '../../images/38593.jpg';
import img91 from '../../images/q9.jpg';
import img01 from '../../images/pw1.jpg';

export const ParallaxSection = styled.section`
  display: flex;
  justify-content: center;
`;

export const ParallaxBackgroundImg1 = styled.div`
  display: block;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  z-index: -1;
  min-width: 50%;
  min-height: 50%;
  background-size: cover;
  background-position: 100%;
  background-repeat: no-repeat;
  position: absolute;
  background-image: url(${img38582});
`;

export const ParallaxBackgroundImg2 = styled.div`
  left: 0;
  top: 100%;
  display: block;
  z-index: -1;
  min-width: 100%;
  min-height: 100%;
  background-size: cover;
  background-position: 100%;
  background-repeat: no-repeat;
  position: absolute;
  background-image: url(${img38593});
`;

export const ParallaxBackgroundImg3 = styled.div`
  left: 0;
  top: 200%;
  display: block;
  z-index: -1;
  min-width: 100%;
  min-height: 100%;
  background-size: cover;
  background-position: 100%;
  background-repeat: no-repeat;
  position: absolute;
  background-image: url(${img91});
`;

export const ParallaxBackgroundImg4 = styled.div`
  left: 0;
  top: 300%;
  display: block;
  z-index: -1;
  min-width: 100%;
  min-height: 100%;
  background-size: cover;
  background-position: 100%;
  background-repeat: no-repeat;
  position: absolute;
  background-image: url(${img01});
`;
