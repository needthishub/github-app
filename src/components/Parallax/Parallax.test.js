import React, {useState as useStateMock} from 'react';
import {shallow} from 'enzyme';
import Parallax from './Parallax';

jest.mock('react', () => ({
  ...jest.requireActual('react'),
  useState: jest.fn(),
}));

describe('<Parallax />', () => {
  const setOffsetYMock = jest.fn();

  beforeEach(() => {
    useStateMock.mockImplementation((init) => [init, setOffsetYMock]);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('should show parallax effect', () => {
    jest.spyOn(React, 'useEffect').mockImplementation((f) => f());
    const wrapper = shallow(<Parallax />);
    expect(wrapper).toMatchSnapshot();
  });
});

/* const events = {
      preventDefault() {},
      scroll: () => setOffsetYMock(),
    }; */

/* jest.spyOn(window, 'addEventListener').mockImplementation((events, handle) => {
      event[events] = handle;
    });
    jest.spyOn(window, 'removeEventListener').mockImplementation((events, handle) => {
      event[events] = handle;
    }); */

// const events = {};

/* jest.spyOn(window, 'addEventListener').mockImplementation((event, handle) => {
      events[event] = handle;
    });
    jest.spyOn(window, 'removeEventListener').mockImplementation((event, handle) => {
      events[event] = handle;
    }); */

// events.scroll({scrollTop: 'Scroll'});
// expect(wrapper).toMatchSnapshot();

// const customEvent = new Event('scroll');
// wrapper.dispatchEvent(customEvent);

// wrapper.find(Parallax).props().onScroll();
// wrapper.simulate('scroll', events);

// wrapper.props().onScroll();
// expect(setOffsetYMock).toHaveBeenCalled();
// expect(wrapper.prop('onScroll')).toHaveBeenCalled();
// wrapper.dispatchEvent(events);
// expect(setOffsetYMock).toHaveBeenCalled();
// expect(adder).toHaveBeenCalled();
// wrapper.unmount();
// expect(remover).toHaveBeenCalled();
