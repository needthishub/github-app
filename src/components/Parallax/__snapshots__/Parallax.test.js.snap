// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Parallax /> should show parallax effect 1`] = `ShallowWrapper {}`;
